controller/scheduler/fleet.py

```diff
@@ -271,14 +271,14 @@ def run(self, name, image, entrypoint, command):  # noqa
             tran = ssh.get_transport()
 
             def _do_ssh(cmd):
-                chan = tran.open_session()
-                # get a pty so stdout/stderr look right
-                chan.get_pty()
-                out = chan.makefile()
-                chan.exec_command(cmd)
-                output = out.read()
-                rc = chan.recv_exit_status()
-                return rc, output
+                with tran.open_session() as chan:
+                    chan.exec_command(cmd)
+                    while not chan.exit_status_ready():
+                        time.sleep(1)
```

Comment:
Although we probably want to be generous and assume `exit_status_ready()` may not return `True` until a long-running task completes, which is this case could be the full 20 minutes we set as the worker timeout. So I'm not sure we should re-implement the same timeout logic here as we have already [here](https://github.com/deis/deis/blob/master/controller/templates/gconf.py#L11).